source/slang/slang-lower-to-ir.cpp

```diff
@@ -498,6 +498,9 @@ struct SharedIRGenContext
     Dictionary<Stmt*, IRBlock*> breakLabels;
     Dictionary<Stmt*, IRBlock*> continueLabels;
 
+    // A reference manager to a ctor's parent
+    Dictionary<FunctionDeclBase*, IRType*> ctorBaseToParentStruct;
```

Comment:
We should be able to get this with `findResultTypeForConstructorDecl`?

Reply:
SemanticsVisitor logic is disjoint from IRGenContext and is not something which can work-without a fully visited SemanticsVisitor?